imap_processing/tests/conftest.py

```diff
@@ -155,3 +159,79 @@ def _unset_metakernel_path(monkeypatch):
     """Temporarily unsets the SPICE_METAKERNEL environment variable"""
     if os.getenv("SPICE_METAKERNEL", None) is not None:
         monkeypatch.delenv("SPICE_METAKERNEL")
+
+
+@pytest.fixture()
+def generate_spin_data():
+    def make_data(start_met: int, end_met: Optional[int] = None) -> pd.DataFrame:
+        """
+        Generate a spin table CSV covering one or more days.
+        Spin table contains the following fields:
+            (
+            spin_number,
+            spin_start_sec,
+            spin_start_subsec,
+            spin_period_sec,
+            spin_period_valid,
+            spin_phas_valid,
+            spin_period_source,
+            thruster_firing
+            )
+        This function creates spin data using start MET and end MET time.
+        Each spin start data uses the nominal 15 second spin period. The spins that
+        occur from 00:00(Mid-night) to 00:10 UTC are marked with flags for
+        thruster firing, invalid spin period, and invalid spin phase.
+        Parameters
+        ----------
+        start_met : int
+            Provides the start time in Mission Elapsed Time (MET).
+        end_met : int
+            Provides the end time in MET. If not provided, default to one day
+            from start time.
+        Returns
+        -------
+        spin_df : pd.DataFrame
+            Spin data.
+        """
+        if end_met is None:
+            # end_time is one day after start_time
+            end_met = start_met + 86400
+
+        # Create spin start second data of 15 seconds increment
+        spin_start_sec = np.arange(start_met, end_met, 15)
```

Comment:
Do we want end_met included?
```suggestion
        spin_start_sec = np.arange(start_met, end_met + 1, 15)
```